sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -58,7 +61,16 @@ def _parse_conn_str(conn_str):
             shared_access_key = value
         elif key.lower() == "entitypath":
             entity_path = value
-    if not all([endpoint, shared_access_key_name, shared_access_key]):
+        elif key.lower() == "sharedaccesssignature":
+            shared_access_signature = value
+            try:
+                # Expiry can be stored in the "se=<timestamp>" clause of the token. ('&'-separated key-value pairs)
+                # type: ignore
+                shared_access_signature_expiry = int(shared_access_signature.split('se=')[1].split('&')[0])
+            except (IndexError, TypeError, ValueError): # Fallback since technically expiry is optional.
+                # An arbitrary, absurdly large number, since you can't renew.
+                shared_access_signature_expiry = int(time.time() * 2)
```

Comment:
I'm not super confident on `large number` being implemented as `time.time() * 2`.

Have we tested this case -- no expiry provided we default to a very large number?

The concern I have here is that as you know our uamqp is built upon C, whether this would lead to integer overflow on certain platforms, making expiry being some negative value.

Reply:
Wisdom beyond your years.  My inclination would be to use some vestige of INT_MAX.  I'll poke around, but shout if you know off the cuff the incantation that I want to properly align with whatever is being done in the AMQP level.

Reply:
Per our OOB discussion: Writing this test ended up exposing the following [uamqp bug](https://github.com/Azure/azure-uamqp-python/issues/170)

Added issue for future investigation, noting this in the test stub that we can reenable once it's fixed.